EIPS/eip-888.md

```diff
@@ -0,0 +1,189 @@
+    /// @dev MUST throw unless _to address is participating.
+    function transfer(address _to, uint256 _amount) external;
+
+    /// @notice Transfer EXP from one address to another.
+    /// @dev MUST throw unless msg.sender is operator.
+    ///  MUST throw unless _to address is participating.
+    ///  MAY throw if _from address is NOT participating.
+    function transferFrom(address _from, address _to, uint256 _amount) external;
+
+    /// @notice Activate or deactivate participation.
+    /// @dev MUST throw unless msg.sender is _participant.
+    /// @param _participant Address opting in or out of participation.
+    /// @param _participation Participation status of _participant.
+    function approve(address _participant, bool _participation) external;
+
+    /// @notice Reassign operator authority.
+    /// @dev MUST throw unless msg.sender is _operator.
+    /// @dev MUST throw unless _operator is participating.
+    /// @param _operator New operator of the smart contract.
+    function setOperator(address _operator) external;
+}
+```
+
+The *metadata extension* is OPTIONAL for ERC-888 smart contracts. This allows an EXP smart contract to be interrogated for its name and description.
+```
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./IERC888.sol";
+
+/// @title ERC-888 EXP Standard, optional metadata extension
+/// @dev See https://eips.ethereum.org/EIPS/EIP-888
+///  Note: the ERC-165 identifier for this interface is ###ERC888Metadata###.
+interface IERC888Metadata is IERC888 {
+    /// @notice A descriptive name for the EXP in this contract.
+    function name() external view returns (string memory);
+
+    /// @notice A one-line description of the EXP in this contract.
+    function description() external view returns (string memory);
+}
+```
+
+## Rationale
+### Approval
+EXP drops SHALL require pre-approval from the delivery address. This ensures the receiver is a consenting participant in the smart contract.
+
+### Mints & Transfers
+EXP mints and transfers SHALL be at the soul discretion of the contract operator. This party may be a sports team coach or a multisig DAO wallet. We decide not to specify how governance occurs, but only *that* governance occurs. This allows for a wider range of potential use cases than optimizing for particular decision-making forms.
+
+ERC-888 standardizes a control mechanism to allocate community recognition without encouraging financialization of that recognition or easily allowing non-contributors to acquire EXP representing contribution. While it does not ensure meritocracy, it opens the door.
+
+### Token Destruction
+EXP SHOULD allow burning tokens by contract operators. If Bob has contributed greatly to the community, but then is caught stealing from Alice, the community may decide this should lower Bob's standing and influence in the community. Again, while this does not ensure an ethical standard within the community, it opens the door.
```

Comment:
How can an operating contract burn tokens from a participant in the current implementation? I agree, this privilege should be given to contract operators. Right now, I only see the possible use of `transferFrom` transferring EXP from the offending participant to the 0x0 address, which would fail if 0x0 is not registered as a participant. Correct me if I'm wrong, but `_participants[0x0] == false`.

Reply:
Thanks for pointing this out. It brings to mind several potential vulnerabilities that I'll correct in the next PR:
- `_operator` must not be initiated as the zero address
- `_participants` must initiate with zero address as a `participant`
- `_operator` must not be reassigned to the zero address
- Zero address must not be removed from `_participants`

My solution seems inelegant. We could alternatively just ensure all the relevant functions ignore 0x0, but that seems even worse. I wonder if there's a better way?